src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -342,6 +345,25 @@ public function iShouldBeOnTheCheckoutCompleteStep(): void
         );
     }
 
+    /**
+     * @Then I should not be able to confirm order because products does not fit :shippingMethod requirements
+     */
+    public function iShouldNotBeAbleToConfirmOrderBecauseDoesNotBelongsToShippingCategory(
+        ShippingMethodInterface $shippingMethod
+    ): void {
+        $this->iProceededWithShippingMethod($shippingMethod);
+
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+
+        Assert::same(
+            $this->responseChecker->getValue($response, 'message'),
+            'Given shipment is not eligible for provided shipping method.'
+        );
+
+        Assert::same($response->getStatusCode(), 500);
```

Comment:
I wonder if `500` status code is an appropriate one

Reply:
`500` should **never** be used by application to inform about something, its meaning is far of the application life & should inform that something went really wrong on level of server architecture (either programming language or server).

This is application information about problem, so it should be `400` (or `409`).

My 2 cents :)